Modify S5594(XML - Android): Improve text understandability (APPSEC-134) (#1433)
SonarSource/rspec: rules/S5594/xml/rule.adoc

Once an Android component has been exported, it can be used by attackers to

> launch malicious actions and might also give access to other components
> otherwise non-exported.

Reviewer: ```suggestion
launch malicious actions and might also give access to other components
that are not exported.
```
"other components otherwise non-exported" could be worded a bit differently

> Once an Android component has been exported, both it and non-exported
> components are exposed for untrusted use (more information about collateral damage on non-exported components
> https://hackerone.com/reports/951691[here].)

Author: ```suggestion
Once an Android component has been exported, it can be used by attackers to
launch malicious actions and might also give access to other components
otherwise non-exported.
```

> https://hackerone.com/reports/951691[here].)

Reviewer: I read through this report briefly, and it does look interesting, but I have a few notes.

* Wrt "both it and non-exported components are exposed for untrusted use": we speak about it as if it is a guarantee, but it is more of an example as what can happen in specific instances. The report requires an exported Activity that also starts other Activities through Intents without doing any verification. IMO we should not put this as the focus of this rule.
* Though the report is very clear on exploitation, it is not very clear on mitigation. Are the mitigations we define here clear in general, and do they prevent from the attacks mentioned in the article?

Author: Oh yeah, good point. Maybe that's actually out of scope here even. WDYT if we change the whole thing to:

```
Once an Android component has been exported, it is exposed to
untrusted use and might also weaken other components.
```

Author: yeah you are right, my version might actually end up too ambiguous

As a result, sensitive user data can be stolen, and components can be launched
unexpectedly.

For this reason, the following components should be protected:

* Providers
* Activities
* Activity-aliases
* Services

To do so, it is recommended to either set `exported` to `false`, add
`android:readPermission` and `android:writePermission` attributes, or add a
`<permission>` tag.

> **Warning**: For Android versions lower than 12, The presence of intent filters will cause ``++exported++`` to be set to

Author: ```suggestion
**Warning**: When targeting Android versions lower than 12, The presence of intent filters will cause ``++exported++`` to be set to
```

> **Warning**: When targeting Android versions lower than 12, The presence of intent filters will cause ``++exported++`` to be set to

Reviewer: ```suggestion
**Warning**: When targeting Android versions lower than 12, the presence of intent filters will cause ``++exported++`` to be set to
```
Small typo

> **Warning**: The presence of intent filters will cause ``++exported++`` to be set to

Author: ```suggestion
**Warning**: For Android versions lower than 12, The presence of intent filters will cause ``++exported++`` to be set to
```

> **Warning**: The presence of intent filters will cause exported to be set to
> true by default.

Reviewer: ```suggestion
**Warning**: The presence of intent filters will cause ``++exported++`` to be set to
``++true++`` by default.
```
For consistency with the rest of the document.
